include/flatbuffers/base.h

```diff
@@ -330,31 +330,30 @@ template<typename T> T EndianSwap(T t) {
   if (sizeof(T) == 1) {   // Compile-time if-then's.
     return t;
   } else if (sizeof(T) == 2) {
-    union { T t; uint16_t i; } u;
-    u.t = t;
+    union { T t; uint16_t i; } u = { t };
```

Comment:
Interesting.. making the assignment an initialization makes the error go away?

Reply:
Without this change MSVC reports: `'flatbuffers::EndianSwap::<unnamed-type-u>::<unnamed-type-u>(void)': attempting to reference a deleted function`

Note EndianSwap is only used on big endian machines by EndianScalar, so I commented the `#if FLATBUFFERS_LITTLEENDIAN` to test it.

Reply:
Ok, but can we then make `EndianScalar` just `T` again? `EndianSwap` is fine since it is only ever used on big endian, but `EndianScalar` should be as obvious a no-op as possible for more naive compilers.

Reply:
Same with `WriteScalar`

Reply:
Unfortunately the const-ref in `WriteScalar` is required otherwise `C2719 't': formal parameter with requested alignment of 8 won't be aligned` will persist, as the if-path will be evaluated even if it is not used.

Reply:
`WriteScalar` with ref-arg will be pessimization.
[Initial plan](https://github.com/google/flatbuffers/pull/5313#issuecomment-528495743)  with `AssertScalarT+MutateStruct` is better solution.

Reply:
Agree with @vglavnyy, lets try the original plan then. `EndianScalar` (and to some extent `WriteScalar`) are such core functions in FlatBuffers, called absolutely everywhere, we can't afford the risk that this will regress performance on some compilers.

Reply:
There is another issue. The `Get` function uses `ReadScalar` via `IndirectHelper`. This of course will break on big endian machines. I thought about intorducing a `GetStruct` function, but then iterators and `operator[]` will be unusable in the current form.

Reply:
Probably, the best solution will be split Array class into two independent classes:
They have different iterators and Read/Write methods.
1. `Array<T>` - current implementation.
2. `ArrayOfStruct<S>` - a new implementation.

@aardappel ?